### ISSUE TITLE
Wrong type for wifiAccess?

### ISSUE THREAD

**Issue by buren** [alexandercerutti/passkit-generator#135]
## Running Node Version

v16.7.0

## Description

Should be able to pass an array, `[{ ssid: "...", password: "..." }]`, but I'm not allowed.

### Current "working" type for passkit-generator
<img width="355" alt="Screen Shot 2023-03-23 at 15 37 47" src="https://user-images.githubusercontent.com/922411/227541115-ca09a4d2-13f5-41cc-a68d-626e6e8ba054.png">

### What the type should be
At least as I can tell this should work, see documentation screenshot below
<img width="785" alt="Screen Shot 2023-03-23 at 15 37 37" src="https://user-images.githubusercontent.com/922411/227541119-dd89c2b2-620d-4d09-85bf-b018a1696705.png">

### Apple documentation
🔗 [Apple docs for semantic tags](https://developer.apple.com/documentation/walletpasses/semantictags)

![Screen Shot 2023-03-24 at 14 57 52](https://user-images.githubusercontent.com/922411/227541108-871dd43a-2d02-4be5-824c-e3bfd5ac4cb8.png)

## Expected behavior

Should be allowed to pass an array. 

### Current "working" type for passkit-generator
<img width="355" alt="Screen Shot 2023-03-23 at 15 37 47" src="https://user-images.githubusercontent.com/922411/227541115-ca09a4d2-13f5-41cc-a68d-626e6e8ba054.png">

### What the type should be
At least as I can tell this should work, see documentation screenshot below
<img width="785" alt="Screen Shot 2023-03-23 at 15 37 37" src="https://user-images.githubusercontent.com/922411/227541119-dd89c2b2-620d-4d09-85bf-b018a1696705.png">

## Other details

🔗 [Apple docs for semantic tags](https://developer.apple.com/documentation/walletpasses/semantictags)
🔗 [Wifi Network semantic tag](https://developer.apple.com/documentation/walletpasses/semantictagtype/wifinetwork)

Love the project ⭐ Thanks a lot! 👑

**Comment by buren** [comment]
Thank you for the quickly reply. I'm a bit confused (haven't use Joi before so bare with me 😅). It seems like `wifiAccess` type [is an array in one place](https://github.com/alexandercerutti/passkit-generator/blob/267d8a6e8c52337f94b7ba75359f8d3cb10a26ae/src/schemas/Semantics.ts#L265) but [not the other](https://github.com/alexandercerutti/passkit-generator/blob/267d8a6e8c52337f94b7ba75359f8d3cb10a26ae/src/schemas/Semantics.ts#L179).

__Object type__
https://github.com/alexandercerutti/passkit-generator/blob/267d8a6e8c52337f94b7ba75359f8d3cb10a26ae/src/schemas/Semantics.ts#L179

__Array type__
https://github.com/alexandercerutti/passkit-generator/blob/267d8a6e8c52337f94b7ba75359f8d3cb10a26ae/src/schemas/Semantics.ts#L265

---

The below diff seems to be working for me locally. Can of course open a PR, but wanted to understand the above first (object vs array in different places).

```diff
diff --git a/src/schemas/Semantics.ts b/src/schemas/Semantics.ts
index e3a8993..9a3a79b 100644
--- a/src/schemas/Semantics.ts
+++ b/src/schemas/Semantics.ts
@@ -176,7 +176,7 @@ export interface Semantics {
 	venuePhoneNumber?: string;
 	venueRoom?: string;
 
-	wifiAccess?: SemanticTagType.WifiNetwork;
+	wifiAccess?: SemanticTagType.WifiNetwork[];
 }
 
 export const Semantics = Joi.object<Semantics>().keys({
```

**Comment by alexandercerutti** [comment]
I gave a quick look to Joi scheme and I saw it wrong in the rush.
So, yeah, it seems to be just an issue with TS.

The patch looks good to me.

> but wanted to understand the above first (object vs array in different places)

I just forgot a couple of squared brackets :D

> Also I had a failed test running the test suite locally with no changes to the code.

Don't worry if you just changed the types.
This packages tests should get completely reworked. They where absolutely my first attempt into doing some unit tests, so they suffer of a series of problems due to my inexperience of the past.

Just to understand, which is the test that is failing?

**Comment by buren** [comment]
## Failed test

```
passkit-generator git:(wifi-access-array-type) npm test                                                                                                                   git:(wifi-access-array-type|)

> passkit-generator@3.1.7 test
> npm run build:spec && npx jasmine


> passkit-generator@3.1.7 build:spec
> rimraf "./spec/*.!(ts)" && npx tsc -p tsconfig.spec.json

spec/FieldsArray.ts:13:8 - error TS2555: Expected at least 3 arguments, but got 2.

 13   fa = new FieldsArray(
           ~~~~~~~~~~~~~~~~
 14    {
    ~~~~
...
 19    pool,
    ~~~~~~~~
 20   );
    ~~~

  lib/FieldsArray.d.ts:12:83
    12     constructor(passInstance: InstanceType<typeof PKPass>, keysPool: Set<string>, fieldSchema: typeof Schemas.Field | typeof Schemas.FieldWithRow, ...args: Schemas.Field[]);
                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'fieldSchema' was not provided.


Found 1 error in spec/FieldsArray.ts:13
```

**Comment by alexandercerutti** [comment]
Ugh, I hate them. I have to put myself into them and really create them again. Just don't worry right now.

**Comment by alexandercerutti** [comment]
Released in [v3.1.8](https://www.npmjs.com/package/passkit-generator/v/3.1.8) 🎉
Thank you for your contribution!
